src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c

```diff
@@ -190,7 +191,22 @@ static attach_state_t ptrace_waitpid(pid_t pid) {
 }
 
 static bool process_doesnt_exist(pid_t pid) {
-  // XXX: This is a stub.
+#if defined(__FreeBSD__)
+  struct kinfo_proc kproc;
+  size_t klen;
+  int mib[] = { CTL_KERN, KERN_PROC, KERN_PROC_PID, pid };
+  if (sysctl(mib, 4, &kproc, &klen, NULL, 0) == -1) {
+    return true;
+  }
+
+  switch (kproc.ki_stat) {
+    case SSTOP:
```

Comment:
The comment on SSTOP in proc.h says "Process debugging or suspension.".  This seems to indicate that the process exists, it is just suspended.  So I think that it is only SZOMB that indicates that it doesn't exist?

Reply:
yes good point.